### ISSUE TITLE
rmarkdown summary crashes R

### ISSUE THREAD

**Issue by gravesti** [rpact-com/rpact#68]
Thanks for the very nice package. I'm working on some reporting integrations and I have encountered crashes with printing markdown summaries. This crashes my whole R session. I have tried with the CRAN version and latest version from github.

```r
design <-getDesignGroupSequential(
  kMax = 3,
  alpha = 0.025,
  beta = 0.2,
  informationRates = c(.4, .8, 1),
  typeOfDesign = "asOF",
  typeBetaSpending = "bsOF"
)

getSimulationSurvival(
  design = design,
  directionUpper = FALSE,
  lambda2 = 0.3,
  hazardRatio = 0.7,
  maxNumberOfSubjects = 500,
  plannedEvents = c(80, 160, 200),
  accrualIntensity = 5
) -> sim_res

print(summary(sim_res), markdown = TRUE)
print(summary(sim_res), markdown = TRUE) #  R crashes the second time it's called
```

### Output

```
> library(rpact)
rpact developer version 4.1.1.9273 loaded
> design <- getDesignGroupSequential(
+   kMax = 3,
+   alpha = 0.025,
+   beta = 0.2,
+   informationRates = c(.4, .8, 1),
+   typeOfDesign = "asOF",
+   typeBetaSpending = "bsOF"
+ )
> getSimulationSurvival(
+   design = design,
+   directionUpper = FALSE,
+   lambda2 = 0.3,
+   hazardRatio = 0.7,
+   maxNumberOfSubjects = 500,
+   plannedEvents = c(80, 160, 200),
+   accrualIntensity = 5
+ ) -> sim_res
> sessionInfo()
R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS 15.1.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rpact_4.1.1.9273

loaded via a namespace (and not attached):
 [1] compiler_4.4.1    R6_2.5.1          fastmap_1.2.0    
 [4] cli_3.6.3         htmltools_0.5.8.1 tools_4.4.1      
 [7] rstudioapi_0.16.0 rappdirs_0.3.3    yaml_2.3.10      
[10] Rcpp_1.0.13-1     rmarkdown_2.28    knitr_1.49       
[13] xfun_0.49         digest_0.6.37     rlang_1.1.4      
[16] evaluate_1.0.1   
> print(summary(sim_res), markdown = TRUE)
[1] "*Simulation of survival data (group sequential design)*\n\n*Design parameters*\n\n* *Information rates*: 0.400, 0.800, 1.000 \n* *Critical values*: 3.357, 2.255, 2.026 \n* *Futility bounds (non-binding)*: 0.168, 1.588 \n* *Cumulative alpha spending*: 0.0003942, 0.0122118, 0.0250000 \n* *Local one-sided significance levels*: 0.0003942, 0.0120779, 0.0213919 \n* *Significance level*: 0.0250 \n* *Test*: one-sided \n\n*User defined parameters*\n\n* *Direction upper*: FALSE \n* *Planned cumulative events*: 80, 160, 200 \n* *Maximum number of subjects*: 500 \n* *Accrual intensity*: 5.0 \n* *Piecewise survival times*: 0.00 \n* *lambda(2)*: 0.300 \n* *Hazard ratio*: 0.700 \n\n*Default parameters*\n\n* *Maximum number of iterations*: 1000 \n* *Seed*: 1453075273 \n* *Planned allocation ratio*: 1 \n* *Conditional power*: NA \n* *Drop-out rate (1)*: 0.000 \n* *Drop-out rate (2)*: 0.000 \n* *Drop-out time*: 12.00 \n* *Theta H0*: 1 \n* *Allocation 1*: 1 \n* *Allocation 2*: 1 \n* *kappa*: 1 \n\n*Results*\n\n* *median(1)*: 3.3 \n* *median(2)*: 2.3 \n* *Accrual time*: 100.00 \n* *lambda(1)*: 0.210 \n* *Analysis time [1]*: 20.01 \n* *Analysis time [2]*: 36.01 \n* *Analysis time [3]*: 43.96 \n* *Expected study duration*: 35.75 \n* *Events not achieved [1]*: 0.000 \n* *Events not achieved [2]*: 0.000 \n* *Events not achieved [3]*: 0.000 \n* *Cumulative number of events [1]*: 80 \n* *Cumulative number of events [2]*: 160 \n* *Cumulative number of events [3]*: 200 \n* *Iterations [1]*: 1000 \n* *Iterations [2]*: 868 \n* *Iterations [3]*: 231 \n* *Overall reject*: 0.6500 \n* *Reject per stage [1]*: 0.0480 \n* *Reject per stage [2]*: 0.4440 \n* *Reject per stage [3]*: 0.1580 \n* *Overall futility stop*: 0.2770 \n* *Futility stop per stage [1]*: 0.0840 \n* *Futility stop per stage [2]*: 0.1930 \n* *Early stop*: 0.7690 \n* *Expected number of subjects*: 178.2 \n* *Expected number of events*: 158.7 \n* *Number of subjects [1]*: 99.6 \n* *Number of subjects [2]*: 179.6 \n* *Number of subjects [3]*: 219.3 \n* *Conditional power (achieved) [1]*: NA \n* *Conditional power (achieved) [2]*: 0.5037 \n* *Conditional power (achieved) [3]*: 0.6179 \n\n*Legend*\n\n* *(i)*: values of treatment arm i\n* *[k]*: values at stage k\n\n\n-----\n\n*Simulation of a survival endpoint*\n\nSequential analysis with a maximum of 3 looks (group sequential design), \none-sided overall significance level 2.5%.\nThe results were simulated for a two-sample logrank test, \nH0: hazard ratio = 1, power directed towards smaller values, \nH1: hazard ratio = 0.7, control lambda(2) = 0.3, \nplanned cumulative events = c(80, 160, 200), maximum number of subjects = 500, \naccrual time = 100, accrual intensity = 5, simulation runs = 1000, \nseed = 1453075273.\n\n| Stage                                |      1 |      2 |      3 |\n| ----- | ----- | ----- | ----- |\n| Planned information rate             |    40% |    80% |   100% |\n| Cumulative alpha spent               | 0.0004 | 0.0122 | 0.0250 |\n| Cumulative beta spent                | 0.0427 | 0.1519 | 0.2000 |\n| Stage levels (one-sided)             | 0.0004 | 0.0121 | 0.0214 |\n| Efficacy boundary (z-value scale)    |  3.357 |  2.255 |  2.026 |\n| Futility boundary (z-value scale)    |  0.168 |  1.588 | |\n| Cumulative power                     | 0.0480 | 0.4920 | 0.6500 |\n| Number of subjects                   |   99.6 |  179.6 |  219.3 |\n| Expected number of subjects under H1 |        |        |  178.2 |\n| Cumulative number of events          |   80.0 |  160.0 |  200.0 |\n| Expected number of events under H1   |        |        |  158.7 |\n| Analysis time                        |  20.01 |  36.01 |  43.96 |\n| Expected study duration under H1     |        |        |  35.75 |\n| Conditional power (achieved)         |        | 0.5037 | 0.6179 |\n| Exit probability for efficacy        | 0.0480 | 0.4440 | |\n| Exit probability for futility        | 0.0840 | 0.1930 | |"
attr(,"class")
[1] "knit_asis"
attr(,"knit_cacheable")
[1] NA
```

**Comment by gravesti** [comment]
The crash happens here:

https://github.com/rpact-com/rpact/blob/ec247e41c57fc746896e07ca41e19d4c5b3ca8dd/R/class_summary.R#L231C1-L237C1

I also get an error message `Error: C stack usage  7953684 is too close to the limit` which suggests a recursion problem. So I am suspicious that this may be creating something circular:
```r
queue[[length(queue) + 1]] <- x$object
attr(x$object, "queue") <- queue
```

**Comment by fpahlke** [comment]
Thanks for the detailled bug report 👍 
The issue was fixed in branch [dev/4.1.1](https://github.com/rpact-com/rpact/tree/dev/4.1.1).
